### PR TITLE
TLS configuration page improvements 

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- fixes overlapping issue in TLS upload page 
-self-signed and CA-signed workflows separated by radio button selection. now only the fields necessary for the selected option are displayed 

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/47643/fix-overlapping-choose-private-key-button-in-kots

#### Special notes for your reviewer:
- Couldn't get it to run following the instructions. had to load the html directly and change the references to the css and js for the screenshots 
Screen Shot 2022-11-18 at 07 27 04
Screen Shot 2022-11-18 at 07 27 16
Screen Shot 2022-11-18 at 07 32 02
Screen Shot 2022-11-18 at 07 32 06

#### Does this PR introduce a user-facing change?

```release-note
- Kurl proxy: Fixes overlapping labels on TLS configuration page 
- Kurl proxy: Self-signed and CA-signed flows are now separated by radio button selection. Now only the fields necessary for the selected option are displayed. 
```

### CODE REVIEW

File: kurl_proxy/assets/tls.html [hunk 1]
```diff
@@ -22,25 +22,43 @@
...
+            </div>
+            <p class="tls-header-sub self-signed-visible">
+              A self-signed TLS certificate is currently used to secure communication
+              between your browser and the admin console. You will see a warning in your browser
+              every time you access the admin console unless you upload your own TLS certificate.
+            </p>
+            <p class="tls-header-sub custom-cert-visible hidden">
+              A CA-signed TLS certificate will be used to secure communication between your browser
+              and the admin console.
+            </p>
```
Comment (rianfowler) [inline kurl_proxy/assets/tls.html:45]: two options for this text

File: kurl_proxy/assets/tls.html [hunk 2]
```diff
@@ -64,11 +82,10 @@
...
 
             <div class="button-wrapper flex alignItems--center">
-              <button class="btn primary" type="submit">Upload &amp; continue</button>
-              <div class="flex-column justifyContent--center">
-                <button id="skip-button" class="text-button" type="submit" formaction="/tls/skip">Skip &amp;
-                  continue</button>
-              </div>
+              <button class="btn primary" type="submit">
+                <span class="self-signed-visible">Continue</span>
+                <span class="custom-cert-visible hidden">Upload &amp; continue</span>
```
Comment (rianfowler) [inline kurl_proxy/assets/tls.html:87]: two options for the button text

File: kurl_proxy/assets/tls.js [hunk 3]
```diff
@@ -1,145 +1,192 @@
...
+
+  function handleTypeToggle(e) {
+    if (e && e.target && e.target.value) {
+      if (e.target.value === "self-signed" || e.target.value === "custom-cert") {
+        toggleLabels();
+      }
+    }
+  }
+};
+startTLS();
```
Comment (rianfowler) [inline kurl_proxy/assets/tls.js:190]: wrapped this in a function so no globals would leak out

File: kurl_proxy/assets/tls.js [hunk 4]
```diff
@@ -1,145 +1,190 @@
...
-  var formData = new FormData();
+      if (useSelfSigned) {
+        skipAndWait(e);
+        return;
+      }
 
-  var certInput = document.getElementById("cert");
-  var keyInput = document.getElementById("key");
-  var hostnameInput = document.getElementById("hostname");
+      uploadAndWait(e);
```
Comment (rianfowler) [inline kurl_proxy/assets/tls.js:23]: now that there is only one button we need to fork the behavior based on the radio button value

**Review by rianfowler — COMMENTED** [review]
it'll be easier to review if you view while hiding whitespace

File: kurl_proxy/assets/tls.js [hunk 5]
```diff
@@ -1,145 +1,190 @@
...
 
-  formData.append("cert", certInput.files[0]);
-  formData.append("key", keyInput.files[0]);
-  formData.append("hostname", hostnameInput.value);
+    var form = document.getElementById("upload-form");
+    if (form) {
+      form.addEventListener("submit", handleSubmit);
+    }
+
+    var skip = document.getElementById("skip-button");
```
Comment (miaawong) [inline kurl_proxy/assets/tls.js:31]: I don't think we have skip-button anymore?

**Review by miaawong — APPROVED** [review]

File: kurl_proxy/assets/tls.js (hunk 5 above)
Comment (rianfowler) [reply kurl_proxy/assets/tls.js:31]: that's true i should probably remove the code that isn't being used

File: kurl_proxy/assets/tls-custom.css [hunk 6]
```diff
@@ -399,6 +399,7 @@ button.text-button:hover {
   overflow: hidden;
   position: absolute;
   z-index: -1;
+  opacity: 0;
```
Comment (rianfowler) [inline kurl_proxy/assets/tls-custom.css:402]: hide the default upload putting. this was manually aligning it so the text from the button would show up to the right of the button. that's not going to work with firefox vs chrome. the spacing is different

File: kurl_proxy/assets/tls-custom.css [hunk 7]
```diff
@@ -435,4 +437,33 @@ button.text-button:hover {
...
+div.hidden {
+  display: none;
+}
+
+p.hidden {
+  display: none;
+}
+#key-label, #cert-label {
+  max-width: 130px;
+  word-wrap: break-word;
```
Comment (rianfowler) [inline kurl_proxy/assets/tls-custom.css:468]: now this wraps the filename under the button

**Review by miaawong — APPROVED** [review]

**Pull request merged**